lib/shelley/test/unit/Cardano/Wallet/Shelley/TransactionSpec.hs

```diff
@@ -2231,34 +2262,45 @@ shrinkTxBody (Cardano.ShelleyTxBody e bod scripts scriptData aux val) = tail
     shrinkUpdates SNothing = []
     shrinkUpdates (SJust _) = [SNothing]
 
+
+balanceTransaction'
+    :: (UTxOIndex, Wallet (SeqState 'Mainnet ShelleyKey), Set Tx)
+    -> StdGenSeed
+    -> PartialTx
+    -> Either ErrBalanceTx SealedTx
+balanceTransaction' wal seed tx  =
+    flip evalRand (stdGenFromSeed seed) $ runExceptT $ balanceTransaction @(Rand StdGen)
+            (Ctx @(Rand StdGen) nullTracer testTxLayer)
+            (delegationAddress @'Mainnet)
+            mockProtocolParametersForBalancing
+            dummyTimeInterpreter
+            wal
+            tx
+
 -- | Tests that 'ErrAssignRedeemersUnresolvedTxIns' can in fact be returned by
 -- 'balanceTransaction'.
 prop_balanceTransactionUnresolvedInputs
     :: Wallet'
     -> ShowBuildable PartialTx
+    -> StdGenSeed
     -> Property
 prop_balanceTransactionUnresolvedInputs
     (Wallet' utxo wal pending)
-    (ShowBuildable partialTx') = withMaxSuccess 400 $
-        forAll (dropResolvedInputs partialTx') $ \(partialTx, dropped) ->
-        not (null dropped) ==>
-            forAllShow (runExceptT $ balanceTransaction
-                (Ctx nullTracer tl)
-                (delegationAddress @'Mainnet)
-                mockProtocolParametersForBalancing
-                dummyTimeInterpreter
-                (utxo, wal, pending)
-                partialTx) (show . Pretty) $ \case
-            Right _
-                -> cover 0 True "success" $ property True
-                   -- Balancing can succeed if the dropped inputs happen to be
-                   -- apart of the wallet UTxO.
-            Left (ErrBalanceTxAssignRedeemers (ErrAssignRedeemersUnresolvedTxIns _))
-                -> cover 1 True "unknown txins" True
-            Left _
-                -> property True
+    (ShowBuildable partialTx')
+    seed = withMaxSuccess 400 $
+        forAll (dropResolvedInputs partialTx') $ \(partialTx, dropped) -> do
+        not (null dropped) ==> do
```

Comment:
Just curious: do we get many discards here when the test is run?

Reply:
Not that many:
```
may fail (1238ms)
        +++ OK, passed 100 tests; 3 discarded (10% unknown txins).
```